package.json

```diff
@@ -40,28 +40,28 @@
   },
   "homepage": "https://github.com/rtfeldman/node-test-runner#readme",
   "dependencies": {
-    "chalk": "3.0.0",
-    "chokidar": "3.4.2",
-    "cross-spawn": "7.0.3",
-    "elmi-to-json": "1.3.0",
-    "fs-extra": "8.1.0",
-    "glob": "7.1.6",
-    "minimist": "1.2.5",
-    "murmur-hash-js": "1.0.0",
-    "split": "1.0.1",
-    "temp": "0.9.1",
-    "which": "2.0.2",
+    "chalk": "^3.0.0",
+    "chokidar": "^3.4.2",
+    "cross-spawn": "^7.0.3",
+    "elmi-to-json": "^1.3.0",
+    "fs-extra": "^8.1.0",
+    "glob": "^7.1.6",
+    "minimist": "^1.2.5",
+    "murmur-hash-js": "^1.0.0",
+    "split": "^1.0.1",
+    "temp": "^0.9.1",
+    "which": "^2.0.2",
     "xmlbuilder": "^15.1.0"
   },
   "devDependencies": {
-    "elm": "0.19.1-3",
-    "elm-format": "0.8.3",
-    "eslint": "7.6.0",
-    "flow-bin": "0.131.0",
-    "mocha": "7.2.0",
-    "prettier": "2.0.5",
-    "shelljs": "0.8.4",
-    "strip-ansi": "6.0.0",
-    "xml2js": "0.4.23"
+    "elm": "latest-0.19.1",
+    "elm-format": "^0.8.3",
+    "eslint": "^7.6.0",
+    "flow-bin": "^0.131.0",
+    "mocha": "^7.2.0",
+    "prettier": "^2.0.5",
```

Comment:
Prettier recommends an exact version. package-lock.json usually takes care of that anyway, but if we ever remove and regenerate it we won’t accidentally cause formatting changes in all files. Up to you if you want to change it or not.

Reply:
Very true, maybe I should pin all dev deps? Do bad dev-deps show up in `npm audit`?

Reply:
```suggestion
    "prettier": "2.0.5",
```

Reply:
I usually pin _all_ my devDeps. Just like I pin all dependencies in an application. I think ranges only ever make sense for sub-dependencies.

`npm audit` checks what’s in your `package-lock.json`:

```
❯ npm audit
npm ERR! code EAUDITNOLOCK
npm ERR! audit Neither npm-shrinkwrap.json nor package-lock.json found: Cannot audit a project without a lockfile
npm ERR! audit Try creating one first with: npm i --package-lock-only

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lydell/.npm/_logs/2020-09-10T10_37_20_134Z-debug.log
```

Your own devDeps are in your package-lock.json, but not your dependencies’ devDeps. So it works like it should: Only contributors of node-test-runner will ever get to know about vulnerabilities in devDeps.

Reply:
brilliant, that is exactly what we want